keps/sig-apps/2255-pod-cost/README.md

```diff
@@ -55,191 +49,210 @@ Items marked with (R) are required *prior to targeting to a milestone / release*
-
-If all pods have the same priority, there is no difference in the normal pod delete decision behaviour. The same applies if the pod-cost annotation is not used at all.
-
-The pod-cost annotation can be changed during operation, for example, if workload changes or a new master gets elected.
+Define a known annotation, namely `controller.kubernetes.io/pod-delete-cost` that 
+applications can set to offer a hint on the cost of deleting a pod compared
+to other pods belonging to the same ReplicaSet. 
 
 ### User Stories (optional)
 
-
 #### Story 1
 
-In an application environment with stateful worker (user-)sessions, it is essential to keep the user sessions alive as good as possible. In case of a scale-down event, the application has to tell the scheduler, which delete decision would have the lowest impact on existing sessions.
+The different pods of an application could have different utilization levels. 
+On scale down, the application may prefer to remove the pods with lower utilization.
 
 #### Story 2
 
-An application consists of identical server processes, but one of the replicas will be the master, which should be kept as long as possible. All other replicas can be treated as cattle workload. Then the master can set the priority annotation with a high priority value as soon as it has finished its startup process. The other replicas can remain either without any priority set, or e.g. with all the same, lower priority. This ensures, that the master replica of this deployment will be protected in a downscale situation.
+On scale down, the application may want to remove pods running on the most expensive 
+nodes first. For example, remove pods from nodes running on standard VMs first
+then from ones running on preemptible/spot VMs (which can be 80% cheaper than standard VMs).
 
 
 ### Risks and Mitigations
 
-On previous Kubernetes ReplicaSet controller versions that don't implement the pod-cost annotation feature, the same application might make false assumptions about the protection of a master instance or workers with open (user-)sessions on it. As the pod-cost annotation would be only a suggestion to the ReplicaSet controller, the application developer should, however, handle the case of a failed master instance or broken user sessions. The feature is just an improvement, not a guarantee, as there might happen timing issues between setting the annotation and the next controller scale-down event.
+One risk is that users perceive the feature as a guarantee to delete order. Documentation
+should stress the fact that this is best effort.
 
 ## Design Details
 
+The pod-delete-cost range will be from [-MaxInt, MaxInt]. The default value is 0.
+Invalid values (like setting the annotation to string) will be converted to
+the default value.
+
+When scaling down a ReplicaSet, controller-manager will prioritize deleting
+pods with lower `pod-delete-cost`. Specifically, the pod-delete-cost will be evaluated after
```

Comment:
This is the only place where pod-delete-cost is wrapped with ``. Is the intention to use `pod-delete-cost` for a value? If not, can you drop the quotation marks?

Reply:
thanks, dropped.